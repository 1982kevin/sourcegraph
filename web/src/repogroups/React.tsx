import { RepogroupMetadata } from './types'
import { SearchPatternType } from '../../../shared/src/graphql/schema'
import * as React from 'react'
export const react: RepogroupMetadata = {
    title: 'React',
    name: 'react',
    url: '/react',
    description:
        "Explore React repositories on GitHub. The search examples show how to use React Hook useState with various data types.",
    examples: [
        {
            title: 'Find imports of useState with regex search',
            exampleQuery: <>import [^;]+useState[^;]+ from 'react'</>,
            rawQuery: "import [^;]+useState[^;]+ from 'react'",
            patternType: SearchPatternType.regexp,
        },
        {
            title: 'Examples of useState with an object as the input parameter',
            description: `useState takes a single argument and can accept a primitive, an array or an object.
            Syntax for usage is [state_value, function_to_update_state_value] = useState (initial_state_value).`,
            exampleQuery: <>{'useState({:[string]})'}</>,
            rawQuery: 'useState({:[string]}) count:1000',
            patternType: SearchPatternType.structural,
        },
        {
            title: 'Examples of useState with any JavaScript data type as the input parameter',
            exampleQuery: <>useState(:[string])</>,
            rawQuery: 'useState(:[string]) count:1000',
            patternType: SearchPatternType.structural,
        },
        {
            title: 'Examples of useState with any JavaScript data type as the input parameter in only TypeScript files',
            exampleQuery: (
                <>
                    useState(:[string])
                    <span className="search-keyword">lang:</span>typescript
                </>
            ),
            rawQuery: 'useState(:[string]) count:1000 lang:typescript',
            patternType: SearchPatternType.structural,
        },
    ],
    homepageDescription: 'Learn React with code examples.',
    homepageIcon: 'https://cdn4.iconfinder.com/data/icons/logos-3/600/React.js_logo-512.png',
}
