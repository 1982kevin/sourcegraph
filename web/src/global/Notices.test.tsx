import React from 'react'
import { Notices } from './Notices'
import * as H from 'history'
import { mount } from 'enzyme'

describe('Notices', () => {
    test('shows notices for location', () =>
        expect(
            mount(
                <Notices
                    location="home"
                    settingsCascade={{
                        subjects: [],
                        final: {
                            notices: [
                                { message: 'a', location: 'home' },
                                { message: 'a', location: 'home', dismissible: true },
                                { message: 'b', location: 'top' },
                            ],
                        },
                    }}
                    history={H.createMemoryHistory()}
                />
            )
        ).toMatchSnapshot())

    test('no notices', () =>
        expect(
            mount(
                <Notices
                    location="home"
                    history={H.createMemoryHistory()}
                    settingsCascade={{ subjects: [], final: { notices: null } }}
                />
            )
        ).toMatchSnapshot())
})
