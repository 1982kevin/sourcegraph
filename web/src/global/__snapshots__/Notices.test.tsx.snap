// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notices no notices 1`] = `null`;

exports[`Notices shows notices for location 1`] = `
<div
  className="notices "
>
  <NoticeAlert
    history="[History]"
    key="0"
    notice={
      Object {
        "location": "home",
        "message": "a",
      }
    }
  >
    <div
      className="alert bg-transparent border "
    >
      <Markdown
        dangerousInnerHTML="<p>a</p>
"
        history="[History]"
      >
        <div
          className="markdown"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<p>a</p>
",
            }
          }
          onClick={[Function]}
        />
      </Markdown>
    </div>
  </NoticeAlert>
  <NoticeAlert
    history="[History]"
    key="1"
    notice={
      Object {
        "dismissible": true,
        "location": "home",
        "message": "a",
      }
    }
  >
    <DismissibleAlert
      className="bg-transparent border "
      partialStorageKey="notice.a"
    >
      <div
        className="alert dismissible-alert bg-transparent border "
      >
        <div
          className="dismissible-alert__content"
        >
          <Markdown
            dangerousInnerHTML="<p>a</p>
"
            history="[History]"
          >
            <div
              className="markdown"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<p>a</p>
",
                }
              }
              onClick={[Function]}
            />
          </Markdown>
        </div>
        <button
          className="btn btn-icon"
          onClick={[Function]}
          type="button"
        >
          <Memo(CloseIcon)
            className="icon-inline"
          >
            <svg
              className="mdi-icon icon-inline"
              fill="currentColor"
              height={24}
              viewBox="0 0 24 24"
              width={24}
            >
              <path
                d="M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z"
              />
            </svg>
          </Memo(CloseIcon)>
        </button>
      </div>
    </DismissibleAlert>
  </NoticeAlert>
</div>
`;
